reactor-core/src/test/java/reactor/core/publisher/FluxConcatMapTest.java

```diff
@@ -380,6 +380,38 @@ public void discardOnError() {
 		            .hasDiscardedExactly("value1", 2, 3); //"value1" comes from error cancelling the only inner in flight, the 2 other values are still raw in the queue
 	}
 
+	@Test
+	public void discardOnInnerErrorWithPrefetch() {
+		StepVerifier.create(Flux.just(1, 2, 3)
+						.concatMap(i -> {
+							if (i == 2) {
+								return Mono.defer(() -> Mono.error(new IllegalStateException("boom: value" + i)));
+							} else {
+								return Mono.defer(() -> Mono.just("value" + i));
+							}
+						}))
+				.expectNext("value1")
+				.expectErrorMessage("boom: value2")
+				.verifyThenAssertThat()
+				.hasDiscardedExactly(3);
+	}
+
+	@Test
+	public void discardOnWithPrefetch() {
```

Comment:
for clarity, rename to `discardOnInnerErrorCallableWithPrefetch`?

Reply:
Done and added two new tests to make sure delayedError version is not discarding any elements. Please take a look